pkgs/build-support/fetchnodemodules/default.nix

```diff
@@ -0,0 +1,102 @@
+{ lib, stdenvNoCC, makeWrapper, nodejs }:
+
+{ src
+, hash ? ""
+, sha256 ? ""
+, name ? null
+, production ? true
```

Comment:
perhaps the default should be false here ?
My impression is that fetchNodeModules will fetch the node_modules. There is therefore a high chance that people will use those dependencies to build something. Building will require dev dependencies. I don't see a context in which it would make sense to not have the dev dependencies in the node_modules (since you won't be able to build the related project with it). Actually if you want to optimize the closure size, you might want to first have a dev dependencies, build the project then replace the node_modules folder with only production dependencies.

Reply:
Yes, the documentation should encourage packagers to run `npm prune --production` after building.

Reply:
ok, then it's a matter of changing what the production flag does. At the moment it only installs dependencies and not devDependencies.